### PR TITLE
Management SDK project template and build infrastructure update

### PR DESCRIPTION
**Pull request by allenjzhang**
1. Added management SDK client project template
2. Adjusted Build.Data to support new management SDK projects
3. Added two inline suppression in Azure.Core since it was giving code analysis warnings

### CODE REVIEW

File: eng/templates/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# DotNet project templates
+
+## Azure.Management SDK client library template
```
Comment (pakrym) [inline eng/templates/README.md:3]: Should this go into the root CONTRIBUTING.md?

File: eng/Packages.Data.props [hunk 2]
```diff
@@ -126,6 +126,7 @@
     
     <!-- Azure SDK packages -->
     <PackageReference Update="Azure.Storage.Blobs" Version="12.0.0" />
+
```
Comment (pakrym) [inline eng/Packages.Data.props:129]: nit: revert

**Review by pakrym — APPROVED** [review]

File: eng/templates/Azure.Management.Template/src/autorest.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Generated code configuration
+
+Run `dotnet msbuild /t:GenerateCode` to generate code.
+
+``` yaml
+input-file:
```
Comment (pakrym) [inline eng/templates/Azure.Management.Template/src/autorest.md:6]: you might want an `azure-arm: true` also added here.

File: eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Version>0.1.0</Version>
```
Comment (weshaggard) [inline eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj:3]: Why version 0.1.0? I'd suggest this be `1.0.0-preview.1` as that is what I'd expect folks to start at.

File: eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    <PackageId>Azure.Management.Template</PackageId>
+    <Description>Azure management client SDK for AzureRP </Description>
+    <PackageTags>azure;management;ProviderNameLowercase</PackageTags>
```
Comment (weshaggard) [inline eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj:6]: I assume things like ProviderNameLowercase is just a placeholder?

File: eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+    <NoWarn>($NoWarn);CA1812;</NoWarn>
```
Comment (weshaggard) [inline eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj:11]: If we need to disable this warning for everyone we should probably also move it into the common props.

File: eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj [hunk 7]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (weshaggard) [inline eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj:1]: While I don't mind a second template I'd like to understand if there is a good reason for it. Can we converge this with the existing template?

File: eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj (hunk 5 above)
Comment (allenjzhang) [reply eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj:6]: Yes, it now takes a single parameter say Microsoft.KeyVault or Microsoft.Network. The keyvault, network (lower case) will show up here.

File: eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj (hunk 7 above)
Comment (allenjzhang) [reply eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj:1]: Are you referring to the SDK\Template folder? Talked to Pavel, the intended purpose is different thus the file content etc. So keeping a separate one. Also that template folder was not set up as dotnet new template.

File: eng/templates/Azure.Management.Template/content/test-resources.json [hunk 8]
```diff
@@ -0,0 +1,4 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0"
```
Comment (pakrym) [inline eng/templates/Azure.Management.Template/content/test-resources.json:3]: This is not enough in me experimentation, resources node is required to be present.

Comment (allenjzhang) [reply eng/templates/Azure.Management.Template/content/test-resources.json]: Added.

File: eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj (hunk 6 above)
Comment (allenjzhang) [reply eng/templates/Azure.Management.Template/src/Azure.Management.Template.csproj]: CA1812 and AZC0012 specific affect mgmt plane right now. I have moved into common props.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**